lib/default.nix

```diff
@@ -2,63 +2,60 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 {
-  self,
   lib,
-  nixpkgs,
+  inputs,
 }: let
-  release = lib.strings.fileContents ../.version;
-  versionSuffix = ".${lib.substring 0 8 (self.lastModifiedDate or self.lastModified or "19700101")}.${self.shortRev or "dirty"}";
-  version = release + versionSuffix;
+  inherit (inputs) nixpkgs;
+  inherit (inputs) nixos-generators;
 in {
-  inherit release versionSuffix version;
   modules = import ./ghaf-modules.nix {inherit lib;};
 
-  # NOTE: Currently supports configuration that generates raw-efi image using nixos-generators
   installer = {
-    systemImgCfg,
+    system,
+    sshKeys,
     modules ? [],
-    userCode ? "",
   }: let
-    inherit (systemImgCfg.nixpkgs.hostPlatform) system;
-
-    pkgs = import nixpkgs {inherit system;};
-
-    installerScript = import ../modules/installer/installer.nix {
-      inherit pkgs;
-      inherit (pkgs) runtimeShell;
-      inherit userCode;
-    };
-
     installerImgCfg = lib.nixosSystem {
```

Comment:
This is not a package, but a function and [module](https://github.com/remimimimimi/ghaf/blob/new-installation-approach/modules/installer/default.nix), for the sake of convenience of Ghaf users. I could’ve done this function as part of a module, but thought that it might be useful in some other cases.

Image generated mainly used to bootstrap installation on hardware that doesn’t have appropriate OS.